app/src/features/apiClient/screens/apiClient/components/clientView/APIClientView.tsx

```diff
@@ -352,6 +353,16 @@ const APIClientView: React.FC<Props> = ({ apiEntry, apiEntryDetails, notifyApiRe
     trackAPIRequestCancelled();
   }, []);
 
+  const handleAuthChange = useCallback(
+    (currentAuthType: AUTHORIZATION_TYPES, updatedKey: string, updatedValue: string, formValues: any) => {
+      setEntry((prevEntry) => {
+        const updatedEntry = updateAuthOptions(prevEntry, currentAuthType, updatedKey, updatedValue, formValues);
```

Comment:
Why are we passing `currentAuthType, updatedKey, updatedValue, formValues` all the way here? Shouldn't the AuthorizationView component handle the preparation of this and pass just the `AuthOptions` to the `handleAuthChange` method directly and this just becomes `entry.auth = value`